datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -460,3 +460,44 @@ select unnest(column1) from (select * from (values([1,2,3]), ([4,5,6])) limit 1
 
 statement ok
 drop table unnest_table;
+
+## unnest list followed by unnest struct
+query ?T?
+select unnest(unnest(column3)), column3 from nested_unnest_table;
+----
+[a] b [{c0: [a], c1: b}] 
+
+## unnest->field_access->unnest
+query T?
+select unnest(unnest(column3)['c0']), column3 from nested_unnest_table;
+----
+a [{c0: [a], c1: b}]
+
+## double unnest list
+query I?
+select unnest(unnest(column2['c2'])), column2 from nested_unnest_table;
```

Comment:
could you also please add tests for:
1. Multiple unnest: `select unnest(col1), unnest(col1)`,  `select unnest(col1), unnest(col12)`
1. Unnest + expression: `select unnest(col1)+1`
2. Unnest on a non structured column (e.g. unnest('foo'))
4. Triple unnest `unnest(unnest(unnnest(..)))`

(it is ok if they error, but I think it would be useful to have the additional test coverage that demonstrates how sophisticated the implementation is)

Reply:
i added more tests